### PR TITLE
Metricbeat docs and Developer Guide

### PR DESCRIPTION
**Pull request by ruflin**

### CODE REVIEW

**Comment by monicasarbu** [comment]
LGTM @dedemorton can you please have a look as it goes into the docs?

File: metricbeat/docs/developer-guide.asciidoc [hunk 1]
```diff
@@ -1,10 +1,212 @@
-= Add a module
+= Metricbeat Contributions
 
-* Basic files
-* Configuration
-* Define MetricSets in advance
+Metricbeat can be used in two ways:
```
Comment (monicasarbu) [inline metricbeat/docs/developer-guide.asciidoc]: I would add here that "Metricbeat is periodically interrogating an external system to fetch key metrics information" to say from the beginning what type of systems can be added to Metricbeat.

**Comment by ruflin** [comment]
@dedemorton Give me some more time this week to extend this PR with more docs. I makes more sense if you review it when there is more content.

File: metricbeat/docs/developer-guide.asciidoc (hunk 1 above)
Comment (ruflin) [inline metricbeat/docs/developer-guide.asciidoc]: +1

**Pull request merged**